metabase/metabase
# Migrate notebook breakout step to MLv2

Resolves #30122
Fixes #29731, fixes #31342, fixes #31310

Migrates notebook editor's `BreakoutStep` component to MLv2. That includes: displaying and managing breakout clauses, managing binning strategies, and temporal buckets for specific columns (numeric, coordinates, date/datetime).

👀  The diff looks pretty huge, but a good amount of it are:
* BE fixes — it was easier to do them on this branch to ensure the fix works
* E2E test fixes — updating element queries and assertions for new UI components using fewer test IDs and not using CSS classes, etc.

### To Verify

For different kinds of questions (structured/native, based on raw data/models/questions, flat/multi-stage):

1. Open `/question/:id/notebook`
2. Play around with the breakout step ("Pick a column to group by")
3. Try out different column types (real/computed, with/without binning, with/without temporal bucketing)
4. Overall ensure the behavior matches `master`
